### PR TITLE
Add tt console docs

### LINKED ISSUE #3734: tt connect interactive console: expanded formatting modes
Product: tt CLI
Since: ?
Audience/target: all
Root document: ? https://www.tarantool.io/en/doc/latest/reference/tooling/interactive_console/
? https://www.tarantool.io/en/doc/latest/reference/tooling/tt_cli/connect/
SME: @ GRISHNOV

# Details

The patchset introduces addition commands for the `tt connect` console.

It adds several common commands that relate to a general functionality:

* `\help`, `?` - to show help message with additional information and
  a list of all commands.
* `\quit`, `\q` - quit from the console. Previously this command worked
  on the `tarantool` side, but now it works on the `tt` side and does
  not require execution rights for a user.

In addition, the patch adds commands for extended output modes:

* `\set output <format>` - set format lua, table, ttable or
  yaml (default)
* `\set table_format <format>` - set table format default, jira
  or markdown
* `\set graphics <false/true>` - disables/enables (default)
  pseudographics for table modes
* `\set table_column_width <width>` - set max column width for
  table/ttable
* `\xw <width>` - set max column width for table/ttable
* `\x` - switches output format cyclically
* `\x[l,t,T,y]` - set output format lua, table, ttable or yaml
* `\x[g,G]` - disables/enables pseudographics for table modes

Supported formats:

1. `yaml` - shows data as a YAML document. The format is
   used now by `tarantool` and `tt`.
2. `lua` - shows data as Lua-compatible structs.
3. `table` - shows data as a pseudo graphical table.
4. `ttable` - the same as the table format, but a result table is
   transposed (rotate data from rows to columns).

Supported table formats (for `table` or `ttable` formats):

1. `default` - shows data as a pseudo graphical table.
2. `markdown` - shows data as a Markdown-compatible table.
3. `jira` - shows data as a Jira-compatible table.

Example:

```
> \set output table
> box.space.customers:select()
+------+-----------+------+
| col1 | col2      | col3 |
+------+-----------+------+
| 1    | Elizabeth | 12   |
+------+-----------+------+
> \xg
> box.space.customers:select()
 col1  col2       col3
 1     Elizabeth  12
 2     Mary       46
 3     David      33
```
Requested by @GRISHNOV in https://github.com/tarantool/tt/commit/5189d4fe7d585845cce6d23a5d4b3bb49b808af0.

# Definition of done

- [ ] Define the relation between tt console and tarantool console to find the proper place for this doc
- [ ] Describe the new formatting capabilities in the console documentation

### LINKED ISSUE #3820: tt connect command has no information about SSL options
**Product:** tt CLI
**Since:** 1.0.0
**Audience/target:** all
**Root document:** https://www.tarantool.io/en/doc/latest/reference/tooling/tt_cli/connect/
**SME:** @ psergee

# Details

Sence version 1.0.0 tt CLI support SSL options for _tt connect_ command.
But this is not mentioned in documentation for this command

# Definition of done

- [ ] Supported SSL options present in documentation

The full list of missing `tt connect` options:

```
$ tt connect --help

  -i, --interactive           enter interactive mode after executing 'FILE'
  -l, --language string       language: lua or sql
  -x, --outputformat string   output format: yaml, lua, table or ttable (default "yaml")
      --sslcafile string      path to a trusted certificate authorities (CA) file
      --sslcertfile string    path to an SSL certificate file
      --sslciphers string     colon-separated (:) list of SSL cipher suites the connection
      --sslkeyfile string     path to a private SSL key file

```

### PR DESCRIPTION
**Pull request by p7nov**
Resolves #3734 
Resolves #3820 

- Add new page [tt interactive console](https://docs.d.tarantool.io/en/doc/gh-3734-tt-connect-console/reference/tooling/tt_cli/tt_interactive_console/) with quick tutorial on using the tt console and commands reference
- Add new `tt connect` options `-l` (input language) and `-x` (output format) to the [tt connect reference page](https://docs.d.tarantool.io/en/doc/gh-3734-tt-connect-console/reference/tooling/tt_cli/connect/#options).
- Add the ``--ssl*`` options to the [tt connect reference page](https://docs.d.tarantool.io/en/doc/gh-3734-tt-connect-console/reference/tooling/tt_cli/connect/#options).
- Add new ``tt connect`` auth methods: URI string and env variables.

Deployment: https://docs.d.tarantool.io/en/doc/gh-3734-tt-connect-console/reference/tooling/tt_cli/tt_interactive_console/

**To cherry pick to 2.11**

### CODE REVIEW

File: doc/reference/tooling/tt_cli/connect.rst [hunk 1]
```diff
@@ -28,6 +28,20 @@ Options
 
     ``-`` – read the script from stdin.
 
+.. option:: -i, --interactive
```
Comment (oleg-jukovec) [inline doc/reference/tooling/tt_cli/connect.rst:31]: There are a bit more `tt connect` flags:

```
      --sslcafile string      path to a trusted certificate authorities (CA) file
      --sslcertfile string    path to an SSL certificate file
      --sslciphers string     colon-separated (:) list of SSL cipher suites the connection
      --sslkeyfile string     path to a private SSL key file
```

Or is this not relevant to the issue?

**Review by oleg-jukovec — APPROVED** [review]
Thank you!

File: doc/reference/tooling/tt_cli/connect.rst (hunk 1 above)
Comment (p7nov) [reply doc/reference/tooling/tt_cli/connect.rst:31]: Thanks, added these options and missing ``tt connect`` auth methods (URI string and env variables).

File: doc/reference/tooling/tt_cli/tt_interactive_console.rst [hunk 2]
```diff
@@ -0,0 +1,261 @@
+.. _tt-interactive-console:
+
+tt interactive console
+======================
+
+The ``tt`` utility features an command-line console that allows executing requests
```
Comment (andreyaksenov) [inline doc/reference/tooling/tt_cli/tt_interactive_console.rst]:
```suggestion
The ``tt`` utility features a command-line console that allows executing requests
```

File: doc/reference/tooling/tt_cli/tt_interactive_console.rst [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+
+The shorthands are:
+
+*  ``\xG`` for ``true``
+*  ``\xg`` for ``false``
+
+\\set table_colum_width WIDTH, \\xw WIDTH
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Set maximum printed width of a table cell content. If the length exceeds this value,
```
Comment (andreyaksenov) [inline doc/reference/tooling/tt_cli/tt_interactive_console.rst]:
```suggestion
Set the maximum printed width of a table cell content. If the length exceeds this value,
```

File: doc/reference/tooling/tt_cli/tt_interactive_console.rst [hunk 4]
```diff
@@ -0,0 +1,261 @@
+.. _tt-interactive-console:
+
+tt interactive console
+======================
+
+The ``tt`` utility features an command-line console that allows executing requests
+and Lua code interactively on the connected Tarantool instances.
+It is similar to the :ref:`Tarantool interactive console <interactive_console>` with
+one key difference: the ``tt`` console allows connecting to any available instance,
+both local and remote. Additionally, it offers more flexible output formatting capabilities.
```
Comment (andreyaksenov) [inline doc/reference/tooling/tt_cli/tt_interactive_console.rst:10]: > both local and remote

We have the [Security](https://www.tarantool.io/en/doc/latest/book/admin/security/) topic (I don't know why it has such a name) that describes the difference between local and remote connections. Most of the contnet might be dropped but the main points are:
- A local connection (`tt connect <instance_name>`) opens a console for the `admin` user.
- A remote connection (`tt connect <iproto_socket/address>`) requires credentials for administration. Otherwise, the `guest` user is used.

So, I'd add the link to this topic or added a couple of sentences about this difference to the `Entering the console` section.

File: doc/reference/tooling/tt_cli/tt_interactive_console.rst [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+        type: unsigned
+      - name: band_name
+        type: string
+      - name: year
+        type: unsigned
+      rows:
+      - [1, 'Roxette', 1986]
+    ...
+
+..  code-block:: console
```
Comment (andreyaksenov) [inline doc/reference/tooling/tt_cli/tt_interactive_console.rst:76]: Let's add a short sentence on what this code snippet does.

File: doc/reference/tooling/tt_cli/tt_interactive_console.rst [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+
+    app:storage001> \set language lua
+    app:storage001> box.space.bands:select { 1 }
+    ---
+    - - [1, 'Roxette', 1986]
+    ...
+
+.. note::
```
Comment (andreyaksenov) [inline doc/reference/tooling/tt_cli/tt_interactive_console.rst:84]: To me, this note looks more important than the info about tab-based autocompletion because it describes the second way on how to change the input. So, I'd write this note as a regular sentence and the info about autocompletion as a note. Just a suggestion, up to you.

File: doc/reference/tooling/tt_cli/tt_interactive_console.rst [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+..  code-block:: console
+
+    app:storage001> box.space.bands:select { }
+    ---
+    - - [1, 'Roxette', 1986]
+      - [2, 'Scorpions', 1965]
+      - [3, 'Ace of Base', 1987]
+    ...
+
+You can switch to alternative output formats -- Lua or human-readable tables --
```
Comment (andreyaksenov) [inline doc/reference/tooling/tt_cli/tt_interactive_console.rst]: Don't know why but `human-readable` sounds a bit weird to me in this context :) YAML also looks pretty human-readable

File: doc/reference/tooling/tt_cli/tt_interactive_console.rst [hunk 8]
```diff
@@ -0,0 +1,261 @@
...
+
+For ``table`` and ``ttable`` output, more customizations are possible with the
+following commands:
+
+*   ``\set table_format`` -- table format: default (pseudographics, or ASCII table), Markdown,
+    or Jira-compatible format.
+*   ``\set grahpics`` -- enable or disable graphics for table cells in the default format.
+*   ``\set table_column_width`` -- maximum column width.
+
+..  code-block:: console
```
Comment (andreyaksenov) [inline doc/reference/tooling/tt_cli/tt_interactive_console.rst]: Maybe, separate code snippets inside the list items above would work better.

File: doc/reference/tooling/tt_cli/tt_interactive_console.rst [hunk 9]
```diff
@@ -0,0 +1,261 @@
...
+Show available keyboard shortcuts.
+
+\\set language {lua|sql}
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Set the input language.
+Possible values:
+
+*   ``lua`` (default)
+*   ``sql``.
```
Comment (andreyaksenov) [inline doc/reference/tooling/tt_cli/tt_interactive_console.rst]: Don't remember our guidelines for lists but I'd remove the dot after `sql`

File: doc/reference/tooling/tt_cli/tt_interactive_console.rst [hunk 10]
```diff
@@ -0,0 +1,261 @@
...
+*   ``ttable`` -- the output is a transposed table where tuples are columns.
+    Shorthand: ``\xT``.
+
+.. note::
+
+    The ``\x`` command switches the output format cyclically in the order
+    ``yaml`` > ``lua`` > ``table`` > ``ttable``.
+
+The format of ``table`` and ``ttable`` output can be adjusted using the ``\set table_format``,
+``\set graphics`, and ``\set table_colum_width`` commands.
```
Comment (andreyaksenov) [inline doc/reference/tooling/tt_cli/tt_interactive_console.rst]:
```suggestion
``\set graphics``, and ``\set table_colum_width`` commands.
```

File: doc/reference/tooling/tt_cli/connect.rst [hunk 11]
```diff
@@ -8,8 +8,8 @@ Connecting to a Tarantool instance
     $ tt connect {URI|INSTANCE} [OPTION ...]
 
 
-``tt connect`` connects to a Tarantool instance by its URI or name specified
-during its startup (``tt start``).
+``tt connect`` connects to a Tarantool instance by its URI or instance name specified
+in the current environment.
 
 Options
```
Comment (andreyaksenov) [inline doc/reference/tooling/tt_cli/connect.rst:14]: Let's expand `-u` and `-p` descriptions a bit. `Username` four times in a row looks a bit weird :)

image

File: doc/reference/tooling/tt_cli/connect.rst [hunk 12]
```diff
@@ -28,6 +28,36 @@ Options
...
+    The output format of the :ref:`tt interactive console <tt-interactive-console>`:
+    ``yaml`` (default), ``lua``, ``table``, ``ttable``.
+
+..  option:: --sslcertfile FILEPATH
+
+    The path to a trusted certificate authorities (CA) file for encrypted connections.
+
+..  option:: --sslcafile FILEPATH
+
+    The path to an SSL certificate file for encrypted connections.
```
Comment (andreyaksenov) [inline doc/reference/tooling/tt_cli/connect.rst:47]: Looks like the descriptions for `--sslcafile` and `--sslcertfile` should be swapped.

File: doc/reference/tooling/tt_cli/connect.rst [hunk 13]
```diff
@@ -38,14 +68,50 @@ You can also connect to instances in the same ``tt`` environment
...
+    $ export TT_CLI_USERNAME=myuser
+    $ export TT_CLI_PASSWORD=p4$$w0rD
+    $ tt connect 192.168.10.10:3301
+
+Encrypted connection
+~~~~~~~~~~~~~~~~~~~~
+
+To connect to instances that use SSL encryption, provide the SSL certificate path and
+other encryption parameters in the options ``--sslcertfile``, ``--sslcafile``,
+``--sslkeyfile``, and ``--sslciphers``.
```
Comment (andreyaksenov) [inline doc/reference/tooling/tt_cli/connect.rst]: Strictly speaking, `--sslcertfile` and `--sslkeyfile` are mandatory. Other parameters are optional.

**Review by andreyaksenov — CHANGES_REQUESTED** [review]
Great work! Please see my comments and suggestions below.

File: doc/reference/tooling/tt_cli/tt_interactive_console.rst (hunk 6 above)
Comment (p7nov) [reply doc/reference/tooling/tt_cli/tt_interactive_console.rst:84]: I'd leave this a note because it's a bit out of the page's context: it's not about working in the console, but an external way to select the language.

**Review by andreyaksenov — APPROVED** [review]
